src/test/java/com/cloudbees/jenkins/support/measures/ChronoTest.java

```diff
@@ -0,0 +1,35 @@
+package com.cloudbees.jenkins.support.measures;
+
+import com.cloudbees.jenkins.support.util.Chrono;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ChronoTest {
+    @Test
+    public void chronoTest() throws Exception {
+        Chrono c = new Chrono("Test Chrono");
+        halt(1000);
+        c.markFromPrevious("After 1s");
+        Assert.assertTrue(c.getMeasure("After 1s") >= 1000);
+
+        halt(3000);
+        c.mark("After 3s", "After 1s", "Test Chrono");
+        Assert.assertTrue(c.getMeasure("After 3s") >= 3000);
+        Assert.assertTrue(c.getMeasures("After 3s").get("Test Chrono") >= 4000);
+
+        c.mark("After 3s from beginning");
+        Assert.assertTrue(c.getMeasure("After 3s from beginning") >= 4000);
+
+        halt(400);
+        c.markFromPrevious("After 400");
+        Assert.assertTrue(c.getMeasure("After 400") >= 400);
+        Assert.assertEquals(c.getMeasure("After 400"), c.getMeasures("After 400").get("After 3s from beginning"));
+
+        System.out.println(c.printMeasures());
+
+    }
+
+    private void halt(long millis) throws InterruptedException {
```

Comment:
Sleep might make more sense. Halt means to stop entirely like shutting down.

Reply:
I'll remove the method and just use the `Thread.sleep` call. The method is not worth it